code/parachain/frame/pablo/src/test/dual_asset_constant_product_tests.rs

```diff
@@ -322,55 +307,197 @@ fn test_add_liquidity_with_disproportionate_amount() {
 #[test]
 fn add_lp_with_min_mint_amount() {
 	new_test_ext().execute_with(|| {
-		let pool_init_config = valid_pool_init_config(
-			&ALICE,
-			BTC,
-			Permill::from_percent(50_u32),
-			USDT,
-			Permill::zero(),
+		next_block::<Pablo, Test>();
+
+		let first_asset = BTC;
+		let second_asset = USDT;
+
+		let pool_init_config = PoolInitConfiguration::DualAssetConstantProduct {
+			owner: ALICE,
+			assets_weights: dual_asset_pool_weights(
+				first_asset,
+				Permill::from_percent(50_u32),
+				second_asset,
+			),
+			fee: Permill::zero(),
+		};
+
+		let init_first_asset_amount = currency::BTC::units(100);
+		let init_second_asset_amount = currency::USDT::units(100);
+		let first_asset_amount = currency::BTC::units(10);
+		let second_asset_amount = currency::USDT::units(10);
+
+		let assets_with_init_amounts = BTreeMap::from([
+			(first_asset, init_first_asset_amount),
+			(second_asset, init_second_asset_amount),
+		]);
+
+		let assets_with_amounts = BTreeMap::from([
+			(first_asset, first_asset_amount),
+			(second_asset, second_asset_amount),
+		]);
+
+		let pool_id = create_pool_from_config(pool_init_config);
+		let lp_token = lp_token_of_pool(pool_id);
+
+		// Mint the tokens
+		assert_ok!(Tokens::mint_into(first_asset, &ALICE, init_first_asset_amount));
+		assert_ok!(Tokens::mint_into(second_asset, &ALICE, init_second_asset_amount));
+
+		// Add the liquidity, min amount = 0
+		Test::assert_extrinsic_event(
+			Pablo::add_liquidity(
+				Origin::signed(ALICE),
+				pool_id,
+				assets_with_init_amounts,
+				0,
+				false,
+			),
+			crate::Event::<Test>::LiquidityAdded {
+				who: ALICE,
+				pool_id: 0,
+				minted_lp: 199_999_999_814_806,
```

Comment:
You can verify this value using the simpler formula number (4) in section 4.2

Reply:
for the majority of tests i did the bare minimum to get them to pass, @PoisonPhang and I will rewrite and clean them up following this PR

Reply:
Right, this is a separate suggestion compared to my other testing comments as this is just a way to verify if a value is correct by deriving it in a different way